### ISSUE TITLE
reopen https://github.com/systemd/systemd/issues/2677

### ISSUE THREAD

**Issue by fermulator** [systemd/systemd#10917]
please re-open #2677

**Comment by yuwata** [comment]
I am not sure, but is it related to #10601?

**Comment by yuwata** [comment]
BTW, I think it is better to move the issue here.

**Comment by yuwata** [comment]
Please provide systemd version and other relevant info.

**Comment by arvidjaar** [comment]
> I am not sure, but is it related to #10601?

No. Is it fundamental problem of current framework - "persistent" name based on on-disk information (partition label, filesystem label, device UUID, whatever) cannot be used to uniquely identify physical disk device, because it will be randomly assigned to one of multiple visible paths to disk. 

There is no way to completely (or even sensibly) solve it given current implementation. Multiple paths to disk are seen as separate devices by kernel; at udev level we have individual events for each device path as if it were separate device; we have no way to delay creation of device aliases until we get better information.

Just like in case of multi-device filesystem this requires some persistent instance (daemon) that aggregates events about disk paths before deciding  what aliases should be created.

If at the time when "persistent" aliases are generated we already have information that disk is part of multipath and is handled by multipath, we may be able to skip generating such aliases for individual paths and only ever do it for virtual multipath device. Unfortunately this again is probably impossible - multipath does not begin processing until it gets notified about new device and it gets notified about new device only after event processing by udev is complete.

**Comment by poettering** [comment]
I see no reason to reopen this. Since a long time we only log the offending message at LOG_DEBUG, i.e. it's not visible unless you explicitly ask for it by turning on debug logging.

The fact that people might use similar labels on different devices is something we cannot change. If this is done, it's not usable to reference devices with, but that's expected really, use something different to identify the devices then.

Anyway, let's close this here. I don't see anything new here that wasn't fixed long ago already through the downgrading of the log message. It appears the reporter experienced the issue with a really old version of systemd, but we only track issues with the two most recent systemd versions here, which are 238 and 239, both of which log this message at debug level.

Or to say this differently: if there really is an issue with 238/239 here (and debug logging is not on), then please file a proper new issue, fill in the submission form comprehensively and we'll look into it. But asking to reopen a long-fixed, long-verified issue is not the way to do it. Thank you for understanding.
